ludwig/schema/trainer.py

```diff
@@ -68,6 +75,12 @@ class BaseTrainerConfig(schema_utils.BaseMarshmallowConfig, ABC):
 class ECDTrainerConfig(BaseTrainerConfig):
     """Dataclass that configures most of the hyperparameters used for ECD model training."""
 
+    def __post_init__(self):
+        if self.compile and not _torch_200:
+            raise ConfigValidationError(
+                "ECD model training requires PyTorch 2.0.0 or higher. Please upgrade PyTorch and try again."
+            )
```

Comment:
I would change this to `compile requires PyTorch 2.0.0 or higher. Please upgrade PyTorch and try again`